worker/worker.py

```diff
@@ -0,0 +1,122 @@
+"""
+Stencila Hub Worker.
+
+A Celery app for running jobs.
+See https://docs.celeryproject.org/en/latest/userguide/tasks.html.
+"""
+
+from contextlib import contextmanager, redirect_stdout, redirect_stderr
+from datetime import datetime
+import io
+import json
+import os
+import sys
+import time
+
+from celery import Celery
+from celery.exceptions import SoftTimeLimitExceeded
+
+from execute.process import ProcessSession
+
+
+# Setup the Celery app
+celery = Celery("worker", broker=os.environ["BROKER_URL"], backend='rpc://')
+celery.conf.update(
+    # By default Celery will keep on trying to connect to the broker forever
+    # This overrides that. Initially try again immediately, then add 0.5 seconds for each
+    # subsequent try (with a maximum of 3 seconds).
+    # See https://github.com/celery/celery/issues/4296
+    broker_transport_options={
+        "max_retries": 10,
+        "interval_start": 0,
+        "interval_step": 0.5,
+        "interval_max": 3,
+    }
+)
+
+#@task_postrun.connect
+def task_postrun_handler(
+    signal, sender, task_id, task, args, kwargs, retval, state, **more
+):
+    """
+    Dispatched after a task is executed.
+
+    Sender is the task object being executed.
+    """
+    log = []
+
+    output = task.output if hasattr(task, "output") else ""
+    for line in output.split("\n"):
+        line = line.strip()
+        if len(line) == 0:
+            continue
+        try:
+            entry = json.loads(line)
+        except:
+            entry = {"message": line}
+        log.append(entry)
+
+    if isinstance(retval, SoftTimeLimitExceeded):
+        result = None
+        state = "TERMINATED"
+    elif isinstance(retval, Exception):
+        result = None
+        log.append(
+            {"time": datetime.utcnow().isoformat(), "level": 0, "message": str(retval)}
```

Comment:
Consider also recording `str(type(retval))` as this can be handy just to see if it's `IndexError`, `ValueError` `OSError` etc. Sometimes the exception `message` property is not that great.